boa/src/value/mod.rs

```diff
@@ -183,15 +183,16 @@ impl Value {
                 for (idx, json) in vs.into_iter().enumerate() {
                     new_obj.set_property(
                         idx.to_string(),
-                        Property::data_descriptor(
+                        DataDescriptor::new(
                             Self::from_json(json, context),
                             Attribute::WRITABLE | Attribute::ENUMERABLE | Attribute::CONFIGURABLE,
                         ),
                     );
                 }
                 new_obj.set_property(
                     "length".to_string(),
```

Comment:
Can you remove the `.to_string()`? I know you didn't change this line, but its an unnecessary allocation.